lib/bootstrap_form/form_builder.rb

```diff
@@ -61,9 +61,18 @@ def initialize(object_name, object, template, options)
 
     def file_field_with_bootstrap(name, options = {})
       prevent_prepend_and_append!(options)
-      options = options.reverse_merge(control_class: 'form-control-file')
+      options = options.reverse_merge(control_class: "custom-file-input")
       form_group_builder(name, options) do
-        file_field_without_bootstrap(name, options)
+        content_tag(:div, :class => "custom-file") do
+          placeholder = options.delete(:placeholder) || "Choose file"
+          placeholder_opts = { class: "custom-file-label" }
+          placeholder_opts[:for] = options[:id] if acts_like_form_tag
```

Comment:
For Rails 5.1, we may need to copy options[:id] to placeholder_opts[:for] in all cases, not just if `acts_like_form_tag`, in case they're using `bootstrap_form_with`, and therefore the helpers aren't generating IDs.